### ISSUE TITLE
[SM6.8] SV_DispatchGrid diagnostics needed for output records

### ISSUE THREAD

**Issue by tex3d** [microsoft/DirectXShaderCompiler#5958]
**Description**
The current version of `DiagnoseDispatchGridSemantics` is used only to check input records when the SV_DispatchGrid field is actively used.  It would also be useful for the producers of such records to apply the same rules and diagnostics.  Note that for output records, you do not know the downstream launch type, so you have to assume SV_DispatchGrid may or may not be present, but if it is, it should be subject to the same rules (only one field with allowed type).

**Steps to Reproduce**
```hlsl
struct OutputRecord {
// expected-error@+1{{SV_DispatchGrid can only be applied to a declaration of unsigned 32 or 16-bit integer scalar, vector, or array up to 3 elements, not 'uint4'}}
  uint4 grid : SV_DispatchGrid;
  float data;
};

[Shader("node")]
[NodeLaunch("broadcasting")]
[NodeDispatchGrid(1,1,1)]
[NumThreads(32, 1, 1)]
void node(NodeOutput<OutputRecord> output) {}
```

**Actual Behavior**
No error produced.

**Environment**
- DXC version: latest `staging-sm-6.8` branch
- Host Operating System: any

(edited based on updated name in [pending PR](https://github.com/microsoft/DirectXShaderCompiler/pull/5953))
